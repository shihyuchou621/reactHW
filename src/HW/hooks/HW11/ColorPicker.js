import React from 'react';
import style from './style.module.css';
export default function ColorPicker(props) {

  const {
    name,
    color,
    value,
    onChange,
  } = props;

  return (
    <div className={style.colorPicker}>
      <div className={style.colorName}>{color}</div>
      <div className={style.colorRange}>
        <input
          data-testid="ColorPicker"
          min="0"
          max="255"
          type="range" // 拉桿
          name={name}
          value={value}
          onChange={onChange} />
      </div>
    </div>
  );
}
