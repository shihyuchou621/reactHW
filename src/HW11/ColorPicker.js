import React, { Component } from 'react';

export default class ColorPicker extends Component {
  render() {

    const {
      name,
      color,
      value,
      onChange,
    } = this.props;

    return (
      <div className="colorPicker">
        <div className="colorName">{color}</div>
        <div className="colorRange">
          <input
            data-testid="ColorPicker"
            min="0"
            max="255"
            type="range"
            name={name}
            value={value}
            onChange={onChange} />
        </div>
      </div>
    );
  }
}
