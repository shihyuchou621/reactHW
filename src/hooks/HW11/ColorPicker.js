import React, { Component } from 'react';

/**
 * props
 * - color: 'red', 'green', 'blue'
 * - value: 0 ~ 255
 * - onChange: function */
export default class ColorPicker extends Component {
  render() {

    const {
      name,
      color,
      value,
      onChange,
    } = this.props;

    return (
      <div className="colorPicker">
        <div className="colorName">{color}</div>
        <div className="colorRange">
          <input
            data-testid="ColorPicker"
            min="0"
            max="255"
            type="range"
            name={name}
            value={value}
            onChange={onChange} />
        </div>
      </div>
    );
  }
}
