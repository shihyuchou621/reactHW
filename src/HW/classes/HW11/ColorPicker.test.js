import React from 'react';

import {
  cleanup,
  render,
  fireEvent,
} from '@testing-library/react';

import ColorPicker from './ColorPicker';

afterEach(cleanup);

test('should change correctly', () => {
  const onChange = jest.fn(e => e.target.value);
  const { container, getByTestId } = render(
    <ColorPicker
      color="red"
      value={128}
      onChange={onChange}
    />
  );
  fireEvent.change(getByTestId('ColorPicker'), { target: {
    value: '20'
  }});
  expect(onChange.mock.results[0].value).toBe("20");
  expect(container).toMatchSnapshot();
});
