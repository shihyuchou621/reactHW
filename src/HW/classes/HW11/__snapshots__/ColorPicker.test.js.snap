// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should change correctly 1`] = `
<div>
  <div
    class="colorPicker"
  >
    <div
      class="colorName"
    >
      red
    </div>
    <div
      class="colorRange"
    >
      <input
        data-testid="ColorPicker"
        max="255"
        min="0"
        type="range"
        value="128"
      />
    </div>
  </div>
</div>
`;
